e2e-tests/gef/cypress/e2e/facility/bank-review-date.spec.js

```diff
@@ -0,0 +1,250 @@
+      aboutFacility.coverStartDateDay().clear().type(tomorrowDay);
+      aboutFacility.coverStartDateMonth().clear().type(tomorrowMonth);
+      aboutFacility.coverStartDateYear().clear().type(tomorrowYear);
+      aboutFacility.coverEndDateDay().clear().type(todayDay);
+      aboutFacility.coverEndDateMonth().clear().type(todayMonth);
+      aboutFacility.coverEndDateYear().clear().type(nextYear);
+      aboutFacility.isUsingFacilityEndDateNo().click();
+
+      aboutFacility.continueButton().click();
+
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      bankReviewDate.bankReviewDateDay().clear().type(todayDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(todayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(todayYear);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary();
+      bankReviewDate.bankReviewDateError();
+
+      bankReviewDate.bankReviewDateDay().clear().type(tomorrowDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(tomorrowMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(tomorrowYear);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary().should('not.exist');
+    });
+
+    it('when cover start date is not given, it validates bank review date is after today', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+      aboutFacility.facilityName().clear().type('Name');
+      aboutFacility.shouldCoverStartOnSubmissionYes().click();
+      aboutFacility.coverEndDateDay().clear().type(todayDay);
+      aboutFacility.coverEndDateMonth().clear().type(todayMonth);
+      aboutFacility.coverEndDateYear().clear().type(nextYear);
+      aboutFacility.isUsingFacilityEndDateNo().click();
+
+      aboutFacility.continueButton().click();
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+      bankReviewDate.bankReviewDateDay().clear().type(yesterdayDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(yesterdayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(yesterdayYear);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary();
+      bankReviewDate.bankReviewDateError();
+
+      bankReviewDate.bankReviewDateDay().clear().type(todayDay);
+      bankReviewDate.bankReviewDateMonth().clear().type(todayMonth);
+      bankReviewDate.bankReviewDateYear().clear().type(todayYear);
+
+      bankReviewDate.continueButton().click();
+      bankReviewDate.errorSummary().should('not.exist');
+    });
+
+    it('validates bank review date is less than 6 years in the future', () => {
```

Comment:
I'm not aware of the existing date functionalities and maybe this is tech debt - could the `6` years be a constant e.g `BANK_REVIEW_DATE_MAXIMUM_YEARS`?

So that if the amount of years changes, it's just the constant that needs updating; Otherwise the test description could become misleading

Reply:
Definitely a good idea, will implement in 7161